ouroboros-consensus/docs/ChainSync.md

```diff
@@ -33,9 +37,40 @@ the upstream node.
 ### Our chain changes
 
 When _our_ chain evolves, we get a new fragment. If our new fragment still
-intersects with their chain, all is well. Otherwise, we initiate the
-intersection finding protocol again, establishing a new fragment for the
-upstream node.
+intersects with their chain, all is well. Otherwise, we throw away their
+fragment, wait for the pipeline to drain, initiate the intersection finding
+protocol again, and establish a new fragment for the upstream node.
+
+This is a simple approach, but we need to argue why simply draining the pipe
+(rather than executing the instructions) and throwing away their fragment is
+not too wasteful.
+
+The number of headers we might potentially have to download again due to not
+executing the drained instructions is limited by the high watermark, which is
+small; this is no big deal. However, the fragment we had before we threw it
+away could potentially be very long, so we have to make a case that throwing
+it away is okay.
+
+The scenario is something like this:
+
+```
+                                k
+                           /~~~~~~~~~~\                                 
+
+..................\........------------ our fragment
```

Comment:
We should clarify that the dashes on this line are the headers that have been added when our chain changed.

Reply:
Discussed this on slack, we'll leave it as it.